examples/DivisorCurve.jl

```diff
@@ -0,0 +1,327 @@
+export AffineCurveDivisor, ProjCurveDivisor, iseffective, multiplicity,
+       divisor
+
+################################################################################
+
+abstract type CurveDivisor end
+
+################################################################################
+# The curve C is assume to be smooth and irreducible, but it is not checked.
+@doc Markdown.doc"""
+    AffineCurveDivisor(C::AffinePlaneCurve{S}, D::Dict{Point{S}, Int}) where S <: FieldElem
+
+Given a curve `C` which is assumed to be smooth and irreducible, return the divisor on the curve `C` defined by `D`.
+"""
+mutable struct AffineCurveDivisor{S <: FieldElem} <: CurveDivisor
```

Comment:
Why mutable? Do you support in-place operations?

Reply:
I forgot to remove mutable, it was coming from my first attempts...